### PR TITLE
Feat: #132598 Add notification in case of running a translated build on windows and macOS

### LINKED ISSUE #132598: Notify users to switch builds when running via rosetta
Issue Type: <b>Bug</b>

Visual Studio shows a notification "Restart Visual Studio Code to apply the latest update." Previously when clicking "Update Now" VSCode would automatically restart. However currently it only exits and doesn't manage to restart itself. Manually starting VSCode after this (running `code .`) starts the current version of VSCode (not the updated version) which again shows the update dialog.

VS Code version: Code 1.59.1 (3866c3553be8b268c8a7f8c0482c0c0177aa8bfa, 2021-08-19T11:53:52.479Z)
OS version: Darwin x64 20.6.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 24)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 3, 3|
|Memory (System)|16.00GB (0.03GB free)|
|Process Argv|. --crash-reporter-id aa9fb884-c292-4430-89f4-feb0adbe6d93|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (27)</summary>

Extension|Author (truncated)|Version
---|---|---
pgformatter|bra|1.17.0
bracket-pair-colorizer-2|Coe|0.2.1
vscode-eslint|dba|2.1.23
xml|Dot|2.5.1
gitlens|eam|11.6.0
EditorConfig|Edi|0.16.4
prettier-vscode|esb|8.1.0
haskell|has|1.6.1
Ionide-fsharp|Ion|5.7.1
language-haskell|jus|3.4.0
theme-karyfoundation-themes|kar|25.0.2
vscode-docker|ms-|1.16.1
csharp|ms-|1.23.15
python|ms-|2021.9.1191016588
jupyter|ms-|2021.8.1236758218
remote-containers|ms-|0.191.1
vscode-typescript-tslint-plugin|ms-|1.3.3
vsliveshare|ms-|1.0.4761
sqltools|mtx|0.23.0
vscode-xml|red|0.18.0
synthwave-vscode|Rob|0.1.8
rust|rus|0.7.8
jinjahtml|sam|0.16.0
vscode-fileutils|sle|3.4.5
ayu|tea|1.0.5
shellcheck|tim|0.16.0
vim|vsc|1.21.7

(7 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspyt653:30270858
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30300192
pythontb:30283811
pythonptprofiler:30281270
vsdfh931cf:30280410
vshan820:30294714
vstes263cf:30335440
pythondataviewer:30285071
pythonvsuse255:30340121
vscod805cf:30301675
pythonvspyt200:30340761
vscextlang:30333561
binariesv615:30325510
pythonvssor306:30344512
bridge0708:30335490
pygetstartedt2:30360495
bridge0723:30353136
pythonf5test824cf:30361778
vsdyn478:30356813
pythonvspyt187:30362782
pydsgst2:30361792
vscus224cf:30358040
vssid140:30362080

```

</details>

### PR DESCRIPTION
**Pull request by antonioprudenzano**
This PR resolve #132598 feature request and adds a info notification when running a non native build on Apple silicon.

### CODE REVIEW

File: src/vs/code/electron-main/app.ts
Comment (deepak1556) [inline src/vs/code/electron-main/app.ts:1221]: You can use [app.runningUnderARM64Translation](https://github.com/electron/electron/blob/main/docs/api/app.md#apprunningunderarm64translation-readonly-macos-windows) which is available from the runtime instead.

**Review by deepak1556 — CHANGES_REQUESTED** [review]
thanks for working on the PR

File: src/vs/code/electron-main/app.ts
Comment (antonioprudenzano) [reply src/vs/code/electron-main/app.ts:1221]: i didn't know that, i'm committing a fix

**Comment by antonioprudenzano** [comment]
Maybe we can add a button to the VSCode download page in the notification?

**Comment by antonioprudenzano** [comment]
@deepak1556 gentle ping, thanks.

**Comment by antonioprudenzano** [comment]
@deepak1556 any news on this PR?

**Comment by deepak1556** [comment]
Sorry for the silence here, I like the idea of having a download link to the universal build in the notification. Can you make the change, if not I can address it in this PR.

**Comment by antonioprudenzano** [comment]
> Sorry for the silence here, I like the idea of having a download link to the universal build in the notification. Can you make the change, if not I can address it in this PR.

Added the button, waiting for review.

File: src/vs/workbench/electron-sandbox/window.ts
Comment (deepak1556) [inline src/vs/workbench/electron-sandbox/window.ts:242]:
```suggestion
                                               const quality = this.productService.quality;
                                               let stableURL = '';
                                               let insidersURL = '';
						if (isMacintosh) {
						       stableURL = 'https://code.visualstudio.com/docs/?dv=osx';
						       insidersURL = 'https://code.visualstudio.com/docs/?dv=osx&build=insiders';
						} else if (isWindows) {
						      stableURL = 'https://code.visualstudio.com/docs/?dv=win32arm64user';
						      insidersURL = 'https://code.visualstudio.com/docs/?dv=win32arm64user&build=insiders';
						}
						this.openerService.open(quality === 'stable' ? stableURL : insidersURL);
```

Comment (antonioprudenzano) [reply src/vs/workbench/electron-sandbox/window.ts:242]: Nevermind app.runningUnderARM64Translation doesn't check for linux translation.

File: src/vs/workbench/electron-sandbox/window.ts
Comment (antonioprudenzano) [reply src/vs/workbench/electron-sandbox/window.ts:234]: Hi,
We could use "You are running a x64 build for PRODUCT_NAME. For better performance download a native PRODUCT_NAME build for your machine.".
We already provided the warning with a button to the correct version of VSCode.

File: src/vs/workbench/electron-sandbox/window.ts
Comment (deepak1556) [inline src/vs/workbench/electron-sandbox/window.ts:234]:
```suggestion
				localize("runningTranslated", "You are running an emulated version of {0}. For better performance download the native arm64 version of {0} build for your machine.", this.productService.nameLong),
```

File: src/vs/workbench/electron-sandbox/window.ts
Comment (isidorn) [reply src/vs/workbench/electron-sandbox/window.ts]: Great. Yeah even better if we know this is the "x64" case.

**Comment by antonioprudenzano** [comment]
I think this PR is ready.
What do you think @deepak1556 ?

**Comment by deepak1556** [comment]
Yup it should be good to go, waiting for another round of review from @isidorn

**Review by isidorn — APPROVED** [review]

**Comment by isidorn** [comment]
Looks great. Thanks for this PR 👏

**Comment by precha1989** [comment]
good

**Comment by deepak1556** [comment]
For visibility, the current notification looks like

Screenshot 2023-06-20 at 14 46 59

**Review by deepak1556 — APPROVED** [review]
Thanks for working on this!

**Pull request merged**
